contribs/optimization/src/main/java/simulatedAnnealing/SimulatedAnnealingConfigGroup.java

```diff
@@ -0,0 +1,201 @@
+
+	@Parameter
+	@Comment("Cooling parameter alpha. Used to control the speed of cooling. Should be >0. " +
+			"Typical values 0.8 <= alpha <= 0.9 for exponential cooling. alpha > 1 for logarithmic cooling.")
+	@PositiveOrZero
+	public double alpha = 0.85;
+
+	@Parameter
+	@Comment("Initial temperature T0, should be >0.")
+	@PositiveOrZero
+	public double initialTemperature = 1000;
+
+	@Parameter
+	@Comment("Number of cooling cycles n.")
+	@PositiveOrZero
+	public int nCoolingCycles = Integer.MAX_VALUE;
+
+	@Parameter
+	@Comment("Final temperature Tn of the system. Should >=0.")
+	@PositiveOrZero
+	public double finalTemperature = 0;
+
+	@Parameter
+	@Comment("Number of simulated annealing iterations (!= MATSim iterations, see iterationRatio) spent at each temperature.")
+	@PositiveOrZero
+	public int iterationsPerTemperature = 3;
+
+	@Parameter
+	@Comment("Number of MATSim iterations for each simulated annealing iteration. This can make sense if other components" +
+			"of MATSim have to react to a new solution first before evaluating it (e.g., DRT rebalancing)")
+	@PositiveOrZero
+	public int iterationRatio = 1;
+
+
+	@Parameter
+	@Comment("Tuning parameter k. See http://doi.org/10.5772/5560. If in doubt, set to 1.")
+	@PositiveOrZero
+	public double k = 1;
+
+	@Parameter
+	@Comment("Reset whenever a new best solution has been found. This may help to widen the search. See https://doi.org/10.3390/math9141625")
+	public boolean resetUponBestSolution = false;
+
+	@Parameter
+	@Comment("Number of iterations without any improvement after which a reset occurs")
+	public int deadEndIterationReset = Integer.MAX_VALUE;
+
+	@Parameter
+	@Comment("Defines whether only temperature, only the current solution or both are reset.")
+	public ResetOption resetOption = ResetOption.temperatureOnly;
+
+
+	@Parameter
+	@Comment("Number of iterations after which there won't be any more resets.")
+	public int lastResetIteration = Integer.MAX_VALUE;
+
+	@Parameter
+	@Comment("Last iteration after which there will be no more optimization and the solution will be fixed to the best" +
+			"found solution.")
+	public int lastIteration = Integer.MAX_VALUE;
+
+
+	public static abstract class PerturbationParams extends ReflectiveConfigGroup implements MatsimParameters {
+
+		public final static String SET_TYPE = "perturbationParams";
+
+		public static final String PERTURBATION_IDENTIFIER = "identifier";
+		public static final String PERTURBATION_WEIGHT = "weight";
+
+
+		private String identifier;
+
+		private double weight;
+
+		public PerturbationParams(String identifier, double weight) {
+			super(SET_TYPE);
+			this.identifier = identifier;
+			this.weight = weight;
+		}
+
+		public PerturbationParams(String identifier) {
+			this(identifier, 1.);
+		}
+
+		/**
+		 * {@value -- PERTURBATION_IDENTIFIER}
+		 */
+		@StringGetter(PERTURBATION_IDENTIFIER)
```

Comment:
I think you can use `@Parameter` and `@Comment` also inside parameter sets. I haven't tried it. If it does not work, then probably we need to fix it somehow.

Reply:
@michalmac Yes, I can confirm this works.

Reply:
Thanks, it's updated now